providers/azureblob/src/main/java/org/jclouds/azureblob/options/CopyBlobOptions.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jclouds.azureblob.options;
+
+import java.util.Map;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableMap;
+
+/**
+ * @see <a href="http://msdn.microsoft.com/en-us/library/dd894037.aspx" />
+ */
+// TODO:
+// x-ms-source-if-modified-since
+// x-ms-source-if-unmodified-since
+// x-ms-source-if-match
+// x-ms-source-if-none-match
+public final class CopyBlobOptions {
+   private final Optional<Map<String, String>> userMetadata;
```

Comment:
This is an interesting corner case -- Azure does not seem to allow replacing user metadata with an empty map?